pkg/registration/register/interface.go

```diff
@@ -71,6 +71,9 @@ type RegisterDriver interface {
 	// InformerHandler returns informer of the related object. If no object needs to be watched, the func could
 	// return nil, nil.
 	InformerHandler(option any) (cache.SharedIndexInformer, factory.EventFilterFunc)
+
+	// ManagedClusterDecorator is to change managed cluster metadata or spec during registration process.
+	ManagedClusterDecorator(managedClusterArn string, managedClusterRoleSuffix string) ManagedClusterDecorator
```

Comment:
in the interface level, the input should not be 
```
managedClusterArn string, managedClusterRoleSuffix string
```
since other driver does not know about this, and if we have another driver later that needs other input, this would need to change the signature of the interface.

The interface should be the same as ManagedClusterDecorator, which is only to mutate the managedcluster.
The managedClusterArn and managedClusterRoleSuffix could be put into the driver at the NewDriver part.